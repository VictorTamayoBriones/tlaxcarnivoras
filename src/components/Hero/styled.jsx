import styled from 'styled-components';
import Fondo from '../../images/heroBG.svg';

const Container = styled.div`
    background-image: url(${Fondo});
    background-repeat: no-repeat;
    background-size: cover;
    background-attachment: fixed;
    height: 100vh;
    padding: 100px 20px 0px 20px;
    display: flex;
    justify-content: space-between;
    img{
        width: 500px;
        height: 500px;
        margin-right: 50px;
    }
    @media (max-width: 930px){
        justify-content: center;
        img{
            width: 250px;
            height: 250px;
            margin: 0;
            position: absolute;
        }
    }
`;

const Datos = styled.div`
    width: 40%;
    height: 100%;
    display: flex;
    flex-direction: column;
    gap: 80px;
    h2{
        color: #fff;
        font-size: 35px;
        letter-spacing: 5px;
    }
    p{
        color: #fff;
        line-height: 30px;
        letter-spacing: 1.5px;
    }

    @media (max-width: 930px){
        width: 100%;
        height: max-content;
        text-align:center;
        position: relative;
        padding: 12px;
        border-radius: 8px;
        z-index: 10;
        background: rgba(0, 0, 0, 0.65);
    }

    @media (max-width: 480px){
        gap: 50px;
        h2{
            font-size: 28px;
        }
        p{
            font-size: 18px;
            line-height: normal;
        }
    }
`;

export { Container, Datos };