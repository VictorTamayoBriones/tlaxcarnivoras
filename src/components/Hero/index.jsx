import styled from 'styled-components';
import Fondo from '../../images/heroBG.svg';

const Hero = () => {
    return (
        <Container/>
    );
}

const Container = styled.div`
    background-image: url(${Fondo});
    background-repeat: no-repeat;
    background-size: cover;
    height: 100vh;
`;

export default Hero;