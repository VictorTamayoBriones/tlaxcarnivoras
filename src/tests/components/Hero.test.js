import React from 'react';
import { shallow } from 'enzyme'
import Hero from '../../components/Hero/index';

describe('Test over the component <Hero/>', () => {
    
    test('should show correctly the component', () => {
        
        const wrapper = shallow( <Hero/> )
        expect( wrapper ).toMatchSnapshot();

    })

})
