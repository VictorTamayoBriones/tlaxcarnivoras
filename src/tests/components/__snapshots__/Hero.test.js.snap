// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test over the component <Hero/> should show correctly the component 1`] = `
<styled.div>
  <styled.div>
    <h2>
      Plantas Carnivoras
    </h2>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Nam quia, quas laborum rem velit sunt ipsum est eveniet sint sed aspernatur enim sit nihil facere, magni facilis totam aliquid ratione odio numquam. Distinctio doloribus laborum ullam magnam? Aspernatur in nemo soluta deleniti porro nostrum optio. Magnam esse accusantium vel? Quo.
    </p>
    <styled.button
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      onClick={[Function]}
      whileHover={
        Object {
          "x": 2,
          "y": -5,
        }
      }
    >
      ver mas
    </styled.button>
  </styled.div>
  <ForwardRef(MotionComponent)
    alt="Tlaxcarnivoras"
    animate={
      Object {
        "x": 0,
      }
    }
    initial={
      Object {
        "x": 1000,
      }
    }
    src="logo.png"
    title="Tlaxcarnivoras"
    transition={
      Object {
        "damping": 40,
        "delay": 0.5,
        "stiffness": 280,
        "type": "spring",
      }
    }
  />
</styled.div>
`;
